nanoFramework.Tools.DebugLibrary.Shared/WireProtocol/Engine.cs

```diff
@@ -4423,6 +4423,58 @@ public int GetPacketMaxLength(Commands.OverheadBase cmd)
             return (int)WireProtocolPacketSize - cmd.Overhead;
         }
 
+        public StorageOperationErrorCodes AddFile(string fileName, byte[] fileContent)
+        {
+            var storop = new Commands.Monitor_StorageOperation();

```

Comment:
We have to create a new Command instance for each one we're sending down the wire protocol. These are added to a queue, assigned an ID and used to track the reply, then removed from the queue and disposed. Reusing them is not super safe...